ex6B/ex6.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Write a string splitter: given a string of comma separated values, would
+ * return an array with the values.
+ *
+ * For example:
+ * split('a,b,c') === ['a', 'b', 'c']
+ * split('hello,world') === ['hello', 'world']
+ * split('hello') === ['hello']
+ * split('hello,,,world') === ['hello', '', '', 'world']
+ * split('') === []
+ *
+ */
+function split(str) {
+
+	var strArray = [], tmpStr = '';
+
+	// If no chars in string no work to do so will return empty array
+	if(str === '') {
+		return [];
+	}
+
+	if (str.indexOf(',', 0) === -1) {
```

Comment:
Try to see what would happen if you remove this if and just leave the else part - I believe it takes care of no comma too.